tests/test_md.py

```diff
@@ -277,6 +282,114 @@ def compute_reference_values():
+        ctxt = custom_ops.Context(coords, v0, box, intg_impl, bps)
+        with pytest.raises(RuntimeError) as e:
+            ctxt.multiple_steps_local(100, local_idxs, radius=radius, temperature=200.0)
+        assert "Local MD temperature didn't match Thermostat's temperature." == str(e.value)
+
+        # Construct context with no potentials, local MD should fail.
+        ctxt = custom_ops.Context(coords, v0, box, intg_impl, [])
+        with pytest.raises(RuntimeError) as e:
+            ctxt.multiple_steps_local(100, local_idxs, radius=radius)
+        assert "unable to find a NonbondedAllPairs potential" == str(e.value)
+
+        # If you have multiple nonbonded potentials, should fail
+        ctxt = custom_ops.Context(coords, v0, box, intg_impl, bps * 2)
+        with pytest.raises(RuntimeError) as e:
+            ctxt.multiple_steps_local(100, local_idxs, radius=radius)
+        assert "found multiple NonbondedAllPairs potentials" == str(e.value)
+
+        ctxt = custom_ops.Context(coords, v0, box, intg_impl, bps)
+        # Run steps of local MD
+        xs, boxes = ctxt.multiple_steps_local(num_steps, local_idxs, store_x_interval=x_interval, radius=radius)
+
+        assert xs.shape[0] == num_steps // x_interval
+        assert boxes.shape[0] == num_steps // x_interval
+
+        # Indices in mol that weren't the last atom should have moved
+        assert np.all(coords[:-1][-(mol.GetNumAtoms() - 1) :] != xs[-1][:-1][-(mol.GetNumAtoms() - 1) :])
+        assert np.any(coords[: -mol.GetNumAtoms()] == xs[-1][: -mol.GetNumAtoms()])
+
+        # Verify that the bound potentials haven't been changed, as local md modifies potentials
+        for ref_val, bp in zip(reference_values, bps):
+            ref_du_dx, ref_u = ref_val
+            test_du_dx, test_u = bp.execute(coords, box)
+            np.testing.assert_array_equal(ref_du_dx, test_du_dx)
+            np.testing.assert_equal(ref_u, test_u)
+
+        group_idxs = get_group_indices(get_bond_list(unbound_potentials[0]))
+
+        pressure = 1.0
+
+        barostat = MonteCarloBarostat(coords.shape[0], pressure, temperature, group_idxs, 1, seed)
+        barostat_impl = barostat.impl(bps)
+
+        intg_impl = intg.impl()
+
+        ctxt = custom_ops.Context(coords, v0, box, intg_impl, bps, barostat=barostat_impl)
+        # Run steps of local MD
+        xs, boxes = ctxt.multiple_steps_local(num_steps, local_idxs, store_x_interval=x_interval, radius=radius)
+
+        assert xs.shape[0] == num_steps // x_interval
+        assert boxes.shape[0] == num_steps // x_interval
+
+        # Running with Barostat should similarly not impact the positions of the global system, as it doesn't
+        # run during local MD
+        assert np.all(coords[:-1][-(mol.GetNumAtoms() - 1) :] != xs[-1][:-1][-(mol.GetNumAtoms() - 1) :])
+        assert np.any(coords[: -mol.GetNumAtoms()] == xs[-1][: -mol.GetNumAtoms()])
+
+        summed_potential = SummedPotential(unbound_potentials, sys_params)
```

Comment:
nit: could this test be split up into several smaller tests? (e.g. separating parameter validation, checks for expected side effects, consistency with and without summed potential, etc.). I think that would make it easier to follow what is being tested.

Reply:
Split up into two tests: https://github.com/proteneer/timemachine/pull/915/commits/2bbc6a37ada043c27f9a49c447fbd31226b2fb6d

Kept the rest of the checks together as they should all be identical